soccer_vision_3d_msgs/scripts/visualizer.py

```diff
@@ -0,0 +1,137 @@
+        self.field_boundary = Marker()
+        self.field_boundary.lifetime = Duration(nanoseconds=self.robot_lifetime).to_msg()
+        self.field_boundary.ns = "rel_field_boundary"
+        self.field_boundary.color = ColorRGBA(g=1.0, a=1.0)
+        self.field_boundary.type = Marker.LINE_STRIP
+        self.field_boundary.action = Marker.MODIFY
+        self.field_boundary.lifetime = Duration(nanoseconds=self.goal_lifetime).to_msg()
+        self.field_boundary.scale = Vector3(x=0.02)
+
+        self.create_subscription(BallArray, "balls_relative", self.balls_cb, 10)
+        self.create_subscription(GoalpostArray, "goal_posts_relative", self.goalpost_cb, 10)
+        self.create_subscription(RobotArray, "robots_relative", self.robot_cb, 10)
+        self.create_subscription(FieldBoundary, "field_boundary_relative", self.field_boundary_cb, 10)
+
+    def conf_to_alpha(self, conf: Confidence):
+        if conf.confidence == Confidence.CONFIDENCE_UNKNOWN:
+            return 1.0
+        return conf.confidence
+
+    def balls_cb(self, msg: BallArray):
+        self.marker_ball.header = msg.header
+        for idx, ball in enumerate(msg.balls):
+            self.marker_ball.pose.position = ball.center
+            self.marker_ball.pose.orientation.w = 1.0
+            self.marker_ball.color.a = self.conf_to_alpha(ball.confidence)
+            self.marker_ball.id = idx
+            self.marker_publisher.publish(self.marker_ball)
+
+    def goalpost_cb(self, msg: GoalpostArray):
+        self.marker_goalpost.header = msg.header
+        for idx, post in enumerate(msg.posts):
+            self.marker_goalpost.pose = post.bb.center
+            self.marker_goalpost.pose.position.z = post.bb.center.position.z + post.bb.size.z / 2
+            self.marker_goalpost.scale = post.bb.size
+            self.marker_goalpost.color.a = self.conf_to_alpha(post.confidence)
+            self.marker_goalpost.id = idx
+            self.marker_publisher.publish(self.marker_goalpost)
+
+    def robot_cb(self, msg: RobotArray):
+        self.marker_robot.header = msg.header
+        for idx, robot in enumerate(msg.robots):
+            self.marker_robot.pose = robot.bb.center
+            self.marker_robot.pose.position.z = robot.bb.center.position.z + robot.bb.size.z / 2
+            self.marker_robot.scale = robot.bb.size
+            if robot.attributes.team == RobotAttribute.TEAM_OWN:
+                self.marker_robot.color.r = 0.0
+                self.marker_robot.color.g = 0.0
+                self.marker_robot.color.b = 1.0
+            elif robot.attributes.team == RobotAttribute.TEAM_OPPONENT:
+                self.marker_robot.color.r = 1.0
+                self.marker_robot.color.g = 0.0
+                self.marker_robot.color.b = 0.0
+            elif robot.attributes.team == RobotAttribute.TEAM_UNKNOWN:
+                self.marker_robot.color.r = 0.0
+                self.marker_robot.color.g = 1.0
+                self.marker_robot.color.b = 0.0
```

Comment:
Why not use `ColorRGBA(...)`? This shortens the code a lot

Reply:
Right, this could be a bit more concise.